### PR TITLE
Git usage and other styleguide updates

### LINKED ISSUE #2496: Styleguide update for IEntity derived classes
IEntity derived classes *must* call the OnDestroyed callback when they are freed or queue freed otherwise things will break

### PR DESCRIPTION
**Pull request by hhyyrylainen**
**Brief Description of What This PR Does**

Styleguide updates and other docs related to git

closes #2496

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: doc/style_guide.md [hunk 1]
```diff
@@ -303,7 +343,19 @@ Godot usage
...
+  a special mechanism, `IEntity` destroyed callbacks)
+
+- Don't use a constructor to setup Godot resources, usually Node
+  derived types should mostly do Godot related, constructor-like
+  operations entirely in _Ready. Many resources are not ready yet when
+  a class is constructed or static variables are being initialized. If
+  this is not followed script variables may not show up correctly in
+  the Godot editor as that relies on creating an instance of the
+  class, and that can fail for example because SimulationParameters
+  are not initialized in the Godot editor.
```
Comment (Kasterisk) [inline doc/style_guide.md:358]: We already have some few Node derived classes that uses constructors e.g `DataPoint`, `IconLegend` and `DataLine` among others, although they don't suffer any issues should their setup be moved into _Ready?

Comment (hhyyrylainen) [reply doc/style_guide.md:358]: Yes, preferably if they can. There *probably* is no issue regarding them as they are dynamically created, however any scene attached objects can cause problems due to the code in the constructor running immediately and not when attaching the scene.

Comment (Kasterisk) [reply doc/style_guide.md:358]: I think it needs to be mentioned that scene attached nodes are especially vulnerable to script variables failing to show up correctly due to instantiation fail.

Comment (hhyyrylainen) [reply doc/style_guide.md:358]: Does it? This is kind of long already and it's mentioned that script variables don't show up. It's kind of implicit that only Godot scene file contained classes can have script variables showing in the first place. How would you nicely add that bit of info to this?

Comment (hhyyrylainen) [reply doc/style_guide.md]: I tried to improve it.

**Review by Kasterisk — APPROVED** [review]

**Pull request merged**